Реализовал отправку сообщений

NadirCianid opened the pull request.

NadirCianid committed:
Реализовал отправку сообщений

RickHammerson commented on src/main/java/faang/school/notificationservice/NotificationServiceApp.java:
```diff
@@ -21,4 +27,15 @@ public static void main(String[] args) {
+
+    @Bean
+    public VonageClient vonageClient() {
+        String apiKey = env.getProperty("vonage.api.key");
```
А почему не вынести переменные в application.yaml с префиксом?

RickHammerson replied:
Ну и разделить создание проперти для него и создание клиента по разным бинам

NadirCianid replied:
Вынес создание бина в отдельный класс

RickHammerson commented on src/main/java/faang/school/notificationservice/service/SmsService.java:
```diff
@@ -0,0 +1,39 @@
+
+        SmsSubmissionResponse response = client.getSmsClient().submitMessage(textMessage);
+
+        if (response.getMessages().get(0).getStatus() == MessageStatus.OK) {
```
А если в списке сообщений будет null? С NPE упадет же

RickHammerson commented on src/main/java/faang/school/notificationservice/service/SmsService.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public void send(UserDto user, String message) {
+        log.info("Sending message: " + message + " To user with id = " + user.getId());
```
У sl4j есть спец знаки, в которые можно вставлять объекты
У String.format( ) - это %s,%d и т.д
У методов log - это {} , то есть можно так 
log.info("Sending message: {} to user with id = {}" ,message ,user.getId())

NadirCianid committed:
Исправил замечания по ревью:
- Добавил проверку на null у SmsSubmissionResponse.getMessages();
- Заменил конкатенацию строк в логе на использование параметров;

NadirCianid committed:
Вынес создание бина VonageClient в отдельный конфигурационный класс для подобных бинов.

NadirCianid committed:
Заменил данные аккаунта Vonage

NadirCianid committed:
Добавил тестовый контроллер

NadirCianid committed:
Добавил тесты и добавил выход из метода при неудаче
